### ISSUE TITLE
[Bug]: substring data type errors

### ISSUE THREAD

**Issue by dengn** [matrixorigin/matrixone#3413]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93): 6dedae20180c9e7ad4159ef0c7ad0f8e8c358dee
- Hardware parameters: MAC M1 ARM 16GB
- OS type: OSX
- Others:
```

### Actual Behavior

Function 'substring' with parameters [BIGINT BIGINT] will be implemented in future version.
Function 'substring' with parameters [BIGINT BIGINT BIGINT] will be implemented in future version.
Function 'substring' with parameters [VARCHAR BIGINT VARCHAR] will be implemented in future version.
Operator '+' with parameters [VARCHAR VARCHAR] will be implemented in future version.

**Comment by daviszhen** [comment]
The first three problems has been solved after #3434 .
The fouth, we will not implement varchar + varchar.

**Comment by daviszhen** [comment]
In pr #3514 

Replace #3413 by #3500,#3506 for some cases
Remove #3413 for some cases
